cookbooks/aws-parallelcluster-environment/files/setup-ephemeral-drives.sh

```diff
@@ -73,7 +72,7 @@ function print_block_device_mapping {
 function check_instance_store {
   if ls /dev/nvme* >& /dev/null; then
     IS_NVME=1
-    MAPPINGS=$(realpath --relative-to=/dev/ -P /dev/disk/by-id/nvme*Instance_Storage* | grep -v "*Instance_Storage*" | uniq)
+    MAPPINGS=$(realpath --relative-to=/dev/ -P /dev/disk/by-id/nvme*Instance_Storage* | grep -v "Instance_Storage" | uniq)
```

Comment:
For this [shellcheck warning SC2063](https://www.shellcheck.net/wiki/SC2063), I deleted the `*`'s as advised, but I am unsure what behavior we want:

Before, the `grep -v "*Instance_Storage*"` wouldn't filter out any results, because the `*` at the beginning wouldn't match anything, so the statement would've been the same as `MAPPINGS=$(realpath --relative-to=/dev/ -P /dev/disk/by-id/nvme*Instance_Storage* | uniq)`. Is that what we want instead?

With the deletion of the `*`'s, `MAPPINGS` would be empty, because `grep` will filter out all results since all the results contain `Instance_Storage`.

Reply:
Let's do this:
`MAPPINGS=$(realpath --relative-to=/dev/ -P /dev/disk/by-id/nvme*Instance_Storage* | grep -v "\*Instance_Storage\*" | uniq)`


Explanation: The command does not filter out any results on instance types without instance store. If the command is run on instance types with instance store, there will be results.

The `-v` in `grep` is revert filter. This is to filter out `/dev/disk/by-id/nvme*Instance_Storage*` from `realpath` when running on instance types without instance store.

My proposal uses `\` to escape `*` to tell shellcheck the `*`s are not Regex keywords

I usually sub-command one by one to understand a command so big :)